cupyx/scipy/spatial/_kdtree.py

```diff
@@ -0,0 +1,273 @@
+        distance_upper_bound : nonnegative float
+            Return only neighbors within this distance.  This is used to prune
+            tree searches, so if you are doing a series of nearest-neighbor
+            queries, it may help to supply the distance to the nearest neighbor
+            of the most recent point.
+
+        Returns
+        -------
+        d : array of floats
+            The distances to the nearest neighbors.
+            If ``x`` has shape ``tuple+(self.m,)``, then ``d`` has shape
+            ``tuple+(k,)``. When k == 1, the last dimension of the output is
+            squeezed. Missing neighbors are indicated with infinite distances.
+        i : ndarray of ints
+            The index of each neighbor in ``self.data``.
+            If ``x`` has shape ``tuple+(self.m,)``, then ``i`` has shape
+            ``tuple+(k,)``. When k == 1, the last dimension of the output is
+            squeezed. Missing neighbors are indicated with ``self.n``.
+
+        Notes
+        -----
+        If the KD-Tree is periodic, the position ``x`` is wrapped into the
+        box.
+
+        When the input k is a list, a query for arange(max(k)) is performed,
+        but only columns that store the requested values of k are preserved.
+        This is implemented in a manner that reduces memory usage.
+
+        Examples
+        --------
+
+        >>> import cupy as cp
+        >>> from cupyx.scipy.spatial import KDTree
+        >>> x, y = cp.mgrid[0:5, 2:8]
+        >>> tree = KDTree(np.c_[x.ravel(), y.ravel()])
+
+        To query the nearest neighbours and return squeezed result, use
+
+        >>> dd, ii = tree.query([[0, 0], [2.2, 2.9]], k=1)
+        >>> print(dd, ii, sep='\n')
+        [2.         0.2236068]
+        [ 0 13]
+
+        To query the nearest neighbours and return unsqueezed result, use
+
+        >>> dd, ii = tree.query([[0, 0], [2.2, 2.9]], k=[1])
+        >>> print(dd, ii, sep='\n')
+        [[2.        ]
+         [0.2236068]]
+        [[ 0]
+         [13]]
+
+        To query the second nearest neighbours and return unsqueezed result,
+        use
+
+        >>> dd, ii = tree.query([[0, 0], [2.2, 2.9]], k=[2])
+        >>> print(dd, ii, sep='\n')
+        [[2.23606798]
+         [0.80622577]]
+        [[ 6]
+         [19]]
+
+        To query the first and second nearest neighbours, use
+
+        >>> dd, ii = tree.query([[0, 0], [2.2, 2.9]], k=2)
+        >>> print(dd, ii, sep='\n')
+        [[2.         2.23606798]
+         [0.2236068  0.80622577]]
+        [[ 0  6]
+         [13 19]]
+
+        or, be more specific
+
+        >>> dd, ii = tree.query([[0, 0], [2.2, 2.9]], k=[1, 2])
+        >>> print(dd, ii, sep='\n')
+        [[2.         2.23606798]
+         [0.2236068  0.80622577]]
+        [[ 0  6]
+         [13 19]]
+        """
+        if self.copy_query_points:
+            if cupy.dtype(x.dtype) is not cupy.dtype(cupy.float64):
```

Comment:
nit: `==` comprehensively compares dtypes.
```suggestion
            if x.dtype == cupy.float64:
```